addons/website_livechat/models/website_visitor.py

```diff
@@ -113,13 +112,10 @@ def _upsert_visitor(self, access_token, force_track_values=None):
         if upsert == 'inserted':
             visitor_sudo = self.sudo().browse(visitor_id)
             if guest := self.env["mail.guest"]._get_guest_from_context():
-                guest_livechats = guest.channel_ids.filtered(lambda c: c.channel_type == "livechat")
-                guest_livechats.channel_ids.livechat_visitor_id = visitor_sudo.id
-                guest_livechats.channel_ids.anonymous_name = (
-                    "Visitor #%d (%s)" % (visitor_sudo.id, visitor_sudo.country_id.name)
-                    if visitor_sudo.country_id
-                    else f"Visitor #{visitor_sudo.id}"
-                )
+                # sudo: mail.guest - guest can access their own channels
+                guest_livechats = guest.sudo().channel_ids.filtered(lambda c: c.channel_type == "livechat")
```

Comment:
I believe this sudo is not correctly placed. Channel ACL allows reading without sudo.
Isn't it needed to write new values instead? Country, visitor?